new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Domain.java

```diff
@@ -5,17 +5,61 @@
 
 import java.util.concurrent.Callable;
 
-public class Domain {
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.ApiextensionsV1Api;
+import io.kubernetes.client.openapi.apis.ApiextensionsV1beta1Api;
+import io.kubernetes.client.openapi.apis.ApisApi;
+import io.kubernetes.client.openapi.apis.CustomObjectsApi;
+import io.kubernetes.client.openapi.models.V1beta1CustomResourceDefinition;
+import oracle.weblogic.kubernetes.extensions.LoggedTest;
 
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class Domain implements LoggedTest {
+
+  private static CustomObjectsApi customObjectsApi = new CustomObjectsApi();
+  private static ApiextensionsV1Api apiextensionsV1Api = new ApiextensionsV1Api();
+  private static ApiextensionsV1beta1Api apiextensionsV1beta1Api = new ApiextensionsV1beta1Api();
+  private static ApisApi apisApi = new ApisApi();
+
+
+  /**
+   * Check if the Domain CRD exists
+   * @return true if domains.weblogic.oracle CRD exists otherwise false
+   * @throws Exception
+   */
+  public static boolean isCRDExists() throws Exception {
+    try {
+      V1beta1CustomResourceDefinition domainBetaCrd =
+          apiextensionsV1beta1Api.readCustomResourceDefinition(
+              "domains.weblogic.oracle", null, null, null);
+      assertNotNull(domainBetaCrd);
+      logger.info("domainBetaCrd is not null");
+      return true;
+    } catch (ApiException aex) {
+      if (aex.getCode() == 404) {
+        assertTrue(false, "Expected CRD domains.weblogic.oracle existed.");
```

Comment:
the message is unclear - it reads like you are saying that the CRD that you expected to exist DOES exist